tools/FeynOpt.hs

```diff
@@ -213,38 +263,51 @@ printHelp = mapM_ putStrLn lines
           ]
 
 
-parseArgs :: [Pass] -> Bool -> Bool -> [String] -> IO ()
-parseArgs passes verify pureCircuit []     = printHelp
-parseArgs passes verify pureCircuit (x:xs) = case x of
+data Options = Options {passes :: [Pass], verify :: Bool, pureCircuit :: Bool, useQASM3 :: Bool}
+
+defaultOptions = Options {passes = [], verify = False, pureCircuit = False, useQASM3 = False}
+
+
+parseArgs :: Bool -> Options -> [String] -> IO ()
+parseArgs doneSwitches options []     = printHelp
```

Comment:
I didn't need to add "doneSwitches" per se, but for completeness' sake I added it since I was in here. Look further down at the handling of the (new) "--" switch: when encountered, any further arguments are interpreted as filenames. "doneSwitches" indicates which mode the parser is in, i.e., are we done parsing switches.